FWCore/Services/plugins/CPU.cc

```diff
@@ -95,26 +83,10 @@ namespace edm {
         s = s.erase(0, s.find_first_not_of(drop));
       }
 
-      std::string eraseExtraSpaces(std::string s) {
-        bool founded = false;
-        std::string aux;
-        for (std::string::const_iterator iter = s.begin(); iter != s.end(); iter++) {
-          if (founded) {
-            if (*iter == ' ')
-              founded = true;
-            else {
-              aux += " ";
-              aux += *iter;
-              founded = false;
-            }
-          } else {
-            if (*iter == ' ')
-              founded = true;
-            else
-              aux += *iter;
-          }
-        }
-        return aux;
+      void compressWhitespace(std::string &s) {
+        auto last = std::unique(
+            s.begin(), s.end(), [](unsigned char a, unsigned char b) -> bool { return std::isspace(a) && a == b; });
```

Comment:
`std::string` uses `char`, and `std::isspace()` takes `int` as an argument, so I wonder if it would safer to just use `auto` argument type?

Otherwise excellent simplification!